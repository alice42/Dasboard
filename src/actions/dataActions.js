export const DATA_REQUEST = 'DATA_REQUEST'
export const DATA_SUCCESS = 'DATA_SUCCESS'
export const DATA_ERROR = 'DATA_ERROR'

export const DATA_REQUESTA = 'DATA_REQUESTA'
export const DATA_SUCCESSA = 'DATA_SUCCESSA'
export const DATA_ERRORA = 'DATA_ERRORA'

export const dataRequest = () => {
  return {
    type: DATA_REQUEST
  }
}

export const dataRequestA = () => {
  return {
    type: DATA_REQUESTA
  }
}
