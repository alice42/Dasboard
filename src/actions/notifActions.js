export const NOTIF_REQUEST = 'NOTIF_REQUEST'
export const NOTIF_SUCCESS = 'NOTIF_SUCCESS'
export const NOTIF_ERROR = 'NOTIF_ERROR'

export const notifRequest = () => {
  return {
    type: NOTIF_REQUEST
  }
}
