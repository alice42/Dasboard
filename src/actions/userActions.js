export const INIT = 'INIT'

export const AUTH_REQUEST = 'AUTH_REQUEST'
export const AUTH_SUCCESS = 'AUTH_SUCCESS'
export const AUTH_ERROR = 'AUTH_ERROR'

export const INFO_SUCCESS = 'INFO_SUCCESS'
export const INFO_REQUEST = 'INFO_REQUEST'
export const INFO_ERROR = 'INFO_ERROR'

export const LOGOUT_REQUEST = 'LOGOUT_REQUEST'
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS'
export const LOGOUT_ERROR = 'LOGOUT_ERROR'

export const init = token => {
  return {
    type: INIT,
    token
  }
}

export const authRequest = data => {
  return {
    type: AUTH_REQUEST,
    data
  }
}

export const logoutRequest = () => {
  return {
    type: LOGOUT_REQUEST
  }
}
