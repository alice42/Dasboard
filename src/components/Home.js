import React, { Component } from 'react'

import BandwidthChart from './BandwidthChart'
import AudienceChart from './AudienceChart'

import TestChart from './TestChart'
import Test2Chart from './Test2Chart'

class Home extends Component {
  componentDidMount() {
    if (!this.props.notif.all) {
      this.props.notifActions.notifRequest()
    }
    if (!this.props.data.bandwidth) {
      this.props.dataActions.dataRequest()
    }
    if (!this.props.data.audience) {
      this.props.dataActions.dataRequestA()
    }
  }
  handleLogout = () => {
    this.props.userActions.logoutRequest()
  }
  render() {
    return (
      <div style={{ width: '100%', display: 'flex', flexDirection: 'column' }}>
        <div>
          notifs
          <div>
            {this.props.notif.all &&
              this.props.notif.all.map((notif, index) => (
                <div key={index}>
                  <div>type: {notif.type}</div>
                  <div>message: {notif.message}</div>
                </div>
              ))}
          </div>
        </div>
        <div style={{ color: 'green' }}>
          Welcome {this.props.user.info.fname} {this.props.user.info.lname}
        </div>
        <div>
          All info
          <div>apitoken: {this.props.user.info.apitoken}</div>
          <div>clientid: {this.props.user.info.clientid}</div>
          <div>company: {this.props.user.info.company}</div>
          <div>description: {this.props.user.info.description}</div>
          <div>email: {this.props.user.info.email}</div>
          <div>timestamp: {this.props.user.info.timestamp}</div>
          <div>
            ({new Date(this.props.user.info.timestamp).toLocaleString()})
          </div>
          <div>website: {this.props.user.info.website}</div>
        </div>
        <button onClick={() => this.handleLogout()}>logout</button>
        <div>
          DATA
          <div>bandwidth in Bits per second</div>
          <div>
            {this.props.data.bandwidth && (
              <BandwidthChart bandwidth={this.props.data.bandwidth} />
            )}
          </div>
          <div>audience in number of viewers</div>
          <div>
            {this.props.data.audience && (
              <AudienceChart viewers={this.props.data.audience} />
            )}
          </div>
          <div>
            {this.props.data.bandwidth && (
              <TestChart bandwidth={this.props.data.bandwidth} />
            )}
          </div>
          <div>
            {this.props.data.bandwidth && (
              <Test2Chart bandwidth={this.props.data.bandwidth} />
            )}
          </div>
        </div>
      </div>
    )
  }
}

export default Home
