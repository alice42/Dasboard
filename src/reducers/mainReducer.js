import { combineReducers } from 'redux'
import userReducer from './userReducer'
import notifReducer from './notifReducer'
import dataReducer from './dataReducer'

const mainReducer = combineReducers({
  user: userReducer,
  notif: notifReducer,
  data: dataReducer
})

export default mainReducer
