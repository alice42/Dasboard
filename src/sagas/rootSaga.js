import { all, fork } from 'redux-saga/effects'
import userSaga from './userSaga'
import notifSaga from './notifSaga'
import dataSaga from './dataSaga'

function* rootSaga() {
  yield all([fork(userSaga), fork(notifSaga), fork(dataSaga)])
}
export default rootSaga
