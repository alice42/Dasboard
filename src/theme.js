import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles'

import AlternateGothic from './assets/fonts/AlternateGothic.woff2'
import SourceSansPro from './assets/fonts/SourceSansPro.woff2'

const headerFont = {
  fontFamily: 'AlternateGothic',
  fontStyle: 'normal',
  src: `
    url(${AlternateGothic}) format('woff2')
  `
}
const bodyFont = {
  fontFamily: 'SourceSansPro',
  fontStyle: 'normal',
  src: `
    url(${SourceSansPro}) format('woff2')
  `
}

let theme = createMuiTheme({
  palette: {
    reds: {
      LightPink: { backgroundColor: '#FFE0E9' },
      HotPink: { backgroundColor: '#C42151' }, // <-- Primary red color (CDN)
      Rouge: { backgroundColor: '#9A193E' },
      Velvet: { backgroundColor: '#570D22' }
    },
    blues: {
      LightBlue: { backgroundColor: '#C6ECFF' },
      Azure: { backgroundColor: '#12A5ED' }, // <-- Primary blue color (P2P)
      PeacockBlue: { backgroundColor: '#005D9F' },
      Twilight: { backgroundColor: '#0A3758' }, // <-- Background color
      TwilightLight: { backgroundColor: '#506A84' }
    },
    greens: {
      LightGreen: { backgroundColor: '#C9EDD9' },
      Green: { backgroundColor: '#3FCB7E' }, // <-- Primay green (Efficiency)
      Viridian: { backgroundColor: '#1D874D' }
    },
    yellows: {
      LightYellow: { backgroundColor: '#FCE4A3' },
      Sunflower: { backgroundColor: '#FFBF10' },
      BurntYellow: { backgroundColor: '#DDA02A' } // <-- Primary yellow color (Audience)
    },
    grayscale: {
      White: { backgroundColor: 'white' },
      OffWhite: { backgroundColor: '#F5F8FA' },
      LightGray: { backgroundColor: '#E1E8ED' },
      Gray: { backgroundColor: '#A1AEB8' },
      Black: { backgroundColor: '#333' }
    }
  },
  status: {
    danger: 'orange'
  },
  typography: {
    h3: {
      fontFamily: 'AlternateGothic'
    },
    h5: {
      fontFamily: 'AlternateGothic'
    },
    body: {
      fontFamily: 'SourceSansPro'
    },
    body1: {
      fontFamily: 'SourceSansPro'
    },
    body2: {
      fontFamily: 'SourceSansPro'
    },
    button: {
      fontFamily: 'SourceSansPro'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '@font-face': [headerFont, bodyFont]
      }
    }
  }
})
theme = responsiveFontSizes(theme)

export default theme
