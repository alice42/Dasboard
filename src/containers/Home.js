import React, { Component } from 'react'
import theme from '../theme'
import {
  ThemeMixinsToolBar,
  StyledContainer,
  StyledMain
} from './styles/StyledHome'

import Header from '../components/Home/Header'
import Content from '../components/Home/Content'

class Home extends Component {
  state = {
    open: { user: false, notif: false },
    anchorEl: { user: null, notif: null },
    value: 0
  }

  componentDidMount() {
    if (!this.props.notif.all) {
      this.props.notifActions.notifRequest()
    }
    if (!this.props.data.bandwidth) {
      this.props.dataActions.dataRequest('bandwidth')
      this.props.dataActions.dataRequest('bandwidth', 'max')
    }
    if (!this.props.data.audience) {
      this.props.dataActions.dataRequest('audience')
    }
    if (!this.props.data.countries) {
      this.props.dataActions.dataRequest('countries')
    }
    if (!this.props.data.platforms) {
      this.props.dataActions.dataRequest('platforms')
    }
  }

  handleLogout = () => {
    this.props.userActions.logoutRequest()
  }

  handleMenu = (e, type) => {
    this.setState({
      open: { ...this.state.open, [`${type}`]: true },
      anchorEl: { ...this.state.anchorEl, [`${type}`]: e.currentTarget }
    })
  }

  handleClose = type => {
    this.setState({
      open: { ...this.state.open, [`${type}`]: false },
      anchorEl: { ...this.state.anchorEl, [`${type}`]: null }
    })
  }

  render() {
    return (
      <StyledContainer
        maxWidth="lg"
        themespacing={theme.spacing(4)}
        spacing={4}
      >
        <Header
          theme={theme}
          handleLogout={this.handleLogout}
          handleClose={this.handleClose}
          handleMenu={this.handleMenu}
          open={this.state.open}
          anchorEl={this.state.anchorEl}
          notifs={this.props.notif.all}
          info={this.props.user.info}
        />

        <StyledMain>
          <ThemeMixinsToolBar theme={theme.mixins.toolbar} />
          <Content
            fetching={this.props.data.fetching}
            bandwidth={this.props.data.bandwidth}
            audience={this.props.data.audience}
            countries={this.props.data.countries}
            platforms={this.props.data.platforms}
          />
        </StyledMain>
      </StyledContainer>
    )
  }
}

export default Home
