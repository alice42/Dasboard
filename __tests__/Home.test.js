import "babel-polyfill"
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render , fireEvent, waitFor, cleanup} from '@testing-library/react'
import { StateMock } from '@react-mock/state';

import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import createSagaMiddleware from 'redux-saga'
import mainReducer from '../src/reducers/mainReducer'
import rootSaga from '../src/sagas/rootSaga'

import * as userActions from '../src/actions/userActions'
import * as notifActions from '../src/actions/notifActions'
import * as dataActions from '../src/actions/dataActions'

import Home from '../src/containers/Home'
import Header from '../src/components/Home/Header'

import theme from '../src/theme'

const sagaMiddleware = createSagaMiddleware()
const middleware = [sagaMiddleware]

composeWithDevTools(applyMiddleware(...middleware))
const store = createStore( 
  mainReducer,
  composeWithDevTools(applyMiddleware(...middleware))
)


afterEach(cleanup)
  describe('default/', () => {
    HTMLCanvasElement.prototype.getContext = jest.fn()
    it('should take a snapshot', () => {
        const props = store.getState()
        props.user.info = {
            clientid: 0,
            company: "company",
            fname: "fname",
            lname: "lname",
            email: "e@mail.com",
            website: "www.website.com",
            timestamp: 112730034000,
            description: "description",
            password: "password",
            apitoken: "apiToken"
        }
      const { asFragment, getByText } = render(<Home 
        {...props}
        dataActions={dataActions}
        notifActions={notifActions}
        userActions={userActions}
      />)

      expect(asFragment(<Home {...props}/>)).toMatchSnapshot()
    })
    it('request if props missing', () => {
      const props = store.getState()
      props.user.info = null 
      const notif = () => { props.notif.all = [{type: 'test', message: 'test'}] }
      const user = () => {
        props.user.info = {
            clientid: 0,
            company: "company",
            fname: "fname",
            lname: "lname",
            email: "e@mail.com",
            website: "www.website.com",
            timestamp: 112730034000,
            description: "description",
            password: "password",
            apitoken: "apiToken"
        } 
      }
      const { container, getByText, getByPlaceholderText, getByTestId } = render(
      <Home {...props}
      dataActions={dataActions} 
      notifActions={notif()}
      userActions={user()}
      />
      )
    })
  })
  
  describe('events handlers /', () => {
    it('open/close menu', async () => {
      const props = store.getState()
      
      const renderComponent = ({ open, anchorEl }) =>
      render(
       <StateMock state={{  
         open,
          anchorEl }}>
          <Home 
          {...props} 
          dataActions={dataActions}
          notifActions={notifActions}
          userActions={userActions}/>
       </StateMock>
    );

    const { getByText, getByPlaceholderText , getByLabelText, getByRole} = renderComponent({  
      open: { user: false, notif: false },
      anchorEl: { user: null, notif: null },
    });
      
    expect(getByLabelText('show new notifications')).toBeInTheDocument();
    fireEvent.click(getByLabelText('show new notifications'));
    fireEvent.click(getByRole('presentation').firstChild);
    fireEvent.click(getByLabelText('account of current user'));
    fireEvent.click(getByRole('presentation').firstChild);
    })

        it('Logout', () => {
       const props = store.getState()
        props.user.info = {
            clientid: 0,
            company: "company",
            fname: "fname",
            lname: "lname",
            email: "e@mail.com",
            website: "www.website.com",
            timestamp: 112730034000,
            description: "description",
            password: "password",
            apitoken: "apiToken"
        }
      const { getByText } = render(<Home
      {...props}
      dataActions={dataActions}
      notifActions={notifActions}
      userActions={userActions}/>
    );
    expect(getByText('Log Out')).toBeInTheDocument(); 
    fireEvent.click(getByText('Log Out'));
    })

  })

