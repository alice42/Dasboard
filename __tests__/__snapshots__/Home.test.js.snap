// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default/ should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root css-uyclst MuiContainer-maxWidthLg"
    spacing="4"
    themespacing="32"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      style="background-color: rgb(10, 55, 88);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
          style="flex-grow: 1;"
        >
          Dashboard
        </h3>
        <div>
          <button
            aria-controls="menu-appbar-notif"
            aria-haspopup="true"
            aria-label="show new notifications"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <span
                class="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
                >
                  0
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="menu-appbar-user"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <main
      class="css-83ceuk"
    >
      <div
        class="css-1jtlko4"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root css-h5c6u5 MuiPaper-elevation1 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root css-h5c6u5 MuiPaper-elevation1 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root css-h5c6u5 MuiPaper-elevation1 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
        >
          <div
            class="MuiPaper-root css-h5c6u5 MuiPaper-elevation1 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
        >
          <div
            class="MuiPaper-root css-h5c6u5 MuiPaper-elevation1 MuiPaper-rounded"
          />
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
